core/modules/css/CssAccess.cc

```diff
@@ -467,74 +467,16 @@ MatchTableParams
 CssAccess::getMatchTableParams(std::string const& dbName,
                                std::string const& tableName) const {
     LOGS(_log, LOG_LVL_DEBUG, "getMatchTableParams(" << dbName << ", " << tableName << ")");
-    _checkVersion();
-
-    std::string const tableKey = _prefix + "/DBS/" + dbName + "/TABLES/" + tableName;
-
-    MatchTableParams params;
-
-    std::vector<std::string> subKeys{"dirTable1", "dirColName1", "dirTable2", "dirColName2", "flagColName"};
-    auto paramMap = _getSubkeys(tableKey + "/match", subKeys);
-    if (paramMap.empty()) {
-        // check table key
-        if (not _kvI->exists(tableKey)) throw NoSuchTable(dbName, tableName);
-        return params;
-    }
-
-    // fill the structure
-    params.dirTable1 = paramMap["dirTable1"];
-    params.dirColName1 = paramMap["dirColName1"];
-    params.dirTable2 = paramMap["dirTable2"];
-    params.dirColName2 = paramMap["dirColName2"];
-    params.flagColName = paramMap["flagColName"];
-
-    return params;
+    TableParams params = getTableParams(dbName, tableName);
+    return params.match;
```

Comment:
Ok, but the comment in the header clearly says "Deprecated: use getTableParams() instead to get a consistent set of all table parameters.", which to me means that these two functions are going away soon. If we want to keep them, then how about we define two private functions

_getPartTableParams(std::string const& dbName, std::string const& tableName, PartTableParams& param);
_getMatchTableParams(std::string const& dbName, std::string const& tableName, MatchTableParams& param);

and then in getTableParams:
TableParams params;
_getPartTableParams(dbName, tableName, params.match);
_getMatchTableParams(dbName, tableName, params.partitioning),


MatchParams mParams;
_getMatchTableParams(dbName, tableName, mParams),

in getMatchTableParams()

and

PartParams pParams;
_getPartTableParams(dbName, tableName, pParams),

in getPartTableParams()